### DISCUSSION TITLE
How do I set my Bandcamp username

### DISCUSSION THREAD

**Discussion by doctorfree** [doctorfree/MusicPlayerPlus#20] [Q&A] [answered]
A Bandcamp username is required to download tracks from bandcamp.com

To configure a Bandcamp username, first locate your Bandcamp username by visiting bandcamp.com and, logging in to your account.

In the upper right corner of the browser window should be a lightning bolt, a heart, and your Bandcamp profile picture. Click on the Bandcamp profile picture and select `settings` from the dropdown menu. On the left side of the Settings display click on `Fan`.

Your Bandcamp username should be displayed on the `Fan` tab of Settings. Copy your Bandcamp username and edit `~/.config/mpprc`.
In `~/.config/mpprc` set `BANDCAMP_USER` to your Bandcamp username.

Any time you make a change to `~/.config/mpprc` you must run `mppinit sync` afterward to synchronize any changes you made out to the appropriate configuration files. In this case, the Calliope configuration file `~/.config/calliope/calliope.conf` must be updated.

After setting the BANDCAMP_USER and running `mppinit sync` you should be able to download your Bandcamp collection with `mppinit bandcamp`.

**Comment by JOJ0** [comment]
I might have read an outdated wiki page and have set my username in this file:
```
$ head /home/jojo/.config/calliope/calliope.conf
[bandcamp]
user = my@band.camp.user
```

But I think the most obvious reason why I configured it in there is the output of the mppinit command:
```
$ mppinit bandcamp                                                       

Downloading albums in your Bandcamp collections
A Bandcamp username must be configured in:
        /home/jojo/.config/calliope/calliope.conf

My Bandcamp username is configured, proceed with download ? (y/n) y

```

**Reply by doctorfree** [reply]
Configuring the Bandcamp username must be done in `~/.config/calliope/calliope.conf` since Calliope is what is being used to download. You can either configure it there directly or configure it in `~/.config/mpprc` and run `mppinit sync`. The `mppinit sync` configures it in `calliope.conf`. So it works either way but in the end only the configuration in `calliope.conf` matters.

I tried to gather up all the various configs from all the components MusicPlayerPlus uses and put them in `~/.config/mpprc` to make it easier to configure things in one place. The `mppinit sync` applies that configuration to all the different config files appropriately.

**Comment by JOJ0** [comment, marked as answer]
Ok it works! Got it! I can't use my email adress as the bandcamp user, it must be the actual username. In the end the username from mpprc is written to calliope.conf. Understood! Works! Thanks a ton! Bandcamp stuff is downloading! :-)

**Comment by doctorfree** [comment]
Perhaps I should document this more clearly. I am marking this as answered and making a note to improve the doc.
